app/models/assignment.rb

```diff
@@ -8,15 +8,33 @@ class Assignment < ApplicationRecord
   }, if: :lti_integrated?
   belongs_to :group
   has_many :projects, class_name: "Project", dependent: :nullify
+  before_destroy :cleanup_notification
 
   after_commit :send_new_assignment_mail, on: :create
   after_commit :set_deadline_job
   after_commit :send_update_mail, on: :update
+  after_create_commit :notify_recipient
 
   enum grading_scale: { no_scale: 0, letter: 1, percent: 2, custom: 3 }
   default_scope { order(deadline: :asc) }
   has_many :grades, dependent: :destroy
 
+  has_noticed_notifications model_name: "NoticedNotification"
+  has_many :noticed_notifications, through: :author
+  has_many :notifications, as: :notifiable  # rubocop:disable Rails/HasManyOrHasOneDependent
```

Comment:
These associations are not needed, 
`has_noticed_notifications model_name: "NoticedNotification"` will do these tasks!
Remove this!